Metadata Update from Wiki (#652, #670) (#344)
artefactual/archivematica-docs: user-manual/metadata/METS.rst

.. _METS_schema:

======================
METS in Archivematica
======================

General description
--------------------

> The Metadata Encoding & Tranmission Standard, or `METS`_, is a schema for

Reviewer: ```suggestion
The Metadata Encoding & Transmission Standard, or `METS`_, is a schema for
```

> encoding various metadata, expressed in XML . It essentially acts a wrapper

Reviewer: ```suggestion
encoding various metadata, expressed in XML. It essentially acts a wrapper
```

around other metadata standards, such as PREMIS and Dublin Core that are used in
Archivematica.

The METS file in Archivematica will have a basic generic structure that is
comprised of the following sections:

> * metsHdr (METS header) (1 only)

Reviewer 2: This is nice, as I skipped below I was thinking about expansion of these terms and then skipped back to the top and saw them!!! 

On the syntax `(1 only)` what do you think of the academic approach to numerals? I believe it's 1 to 9 are written one to nine, and then after 10 digits. I don't know our approach across the larger body of docs. I don't think we have it in the style guide at all [ref](https://github.com/artefactual/archivematica-docs/blob/1.11/style_guide.md).

Reviewer: I don't think there is a consistent convention in place. The [Purdue OWL](https://owl.purdue.edu/owl/general_writing/grammar/writing_numbers.html) states "most people spell out numbers that can be expressed in one or two words and use figures for numbers that are three or more words long." I'd probably suggest that approach if we want to include it in the style guide, as it seems to track well for reading purposes (see the link for examples.)

Author: I will attempt to go for the Purdue OWL's recommendation then as I edit. Maybe we can consider putting this in the wiki style guide?

Reviewer: FYI it's now in the [style guide](https://github.com/artefactual/archivematica-docs/blob/1.11/style_guide.md#formatting-and-punctuation)!

Author: Love this @sallain! Thanks!

> * dmdSec (descriptive metadata section) (1 or more)

Reviewer 2: it looks like we may be missing `amdsec` in this list where it is used below?

> * dmdSec (descriptive metadata section) (1 or more)

Reviewer 2: It looks like we may be missing amdsec in this list where it is used below?

<metsHdr>
^^^^^^^^^

There will be one METS header section for each METS file. It will contain a
CREATEDATE attribute.

.. code:: xml

  <metsHdr CREATEDATE=”2019-06-18T23:52:11”/>

If the AIP has been reingested, the metsHder section will also contain a
LASTMODDATE.

.. code:: xml

  <metsHdr CREATEDATE=”2019-06-18T23:52:11” LASTMODDATE=”2019-06-27T00:28:56”/>

.. _dmdsec:

<dmdSec>
^^^^^^^^

There will be one dmdSec for the whole AIP, which contains PREMIS Intellectual
Entity information.

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 15-35

There may also be one or more dmdSecs for descriptive metadata. The dmdSecs are
numbered dmdSec_1, dmdSec_2, etc. and contain Dublin Core metadata as a default.

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 64-91

MDTYPE will be **DC** for Dublin Core metadata; if another metadata standard is
included then the MDTYPE will be **OTHER**. 

If the user does not enter any DC metadata during transfer/ingest and no DC
metadata was included with the transfer (e.g. as a metadata.csv file, see
:ref:`Import metadata <import-metadata>`), there will be no additional dmdSecs.

.. _amdsec:

<amdSec>
^^^^^^^^^

> There is one amdSec for each object and wraps PREMIS entities as follows:

Reviewer 2: I can't remember without looking, will there be one for each original object and preservation object? Is it worth adding that clarification and/or detail do you think?

Author: yes, I will expand on that for clarity.

* PREMIS Object: techMD (tehnical metadata)
* PREMIS Event: digiprovMD (digital provenance metadata)
* PREMIS Agent: digiprovMD (digital provenance metadata)
* PREMIS Rights: rightsMD (rights metadata)

Each amdSec will include one techMD. The techMD is used to wrap PREMIS Object
semantic units, such as objectIdentifier, objectCharacteristics and
originalName. See the example techMD below:

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 212-258

Each amdsec will include multiple digiprovMDs for PREMIS events related to the
PREMIS Object. Below is an example of a PREMIS event for a fixity check:

   .. literalinclude:: scripts/METS.xml
      :language: xml
      :lines: 820-854

An amdSec for an original object may also contain one or more rightsMDs. The
rightsMD may contain a reference to metadata in another file, such as rights.csv
uploaded with a transfer. Below is an example of a PREMIS Rights statement in
rightsMD section of the AIP METS file:

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 2073-2108

> An amdSec will also have one ore more sourceMD sections. This is the only

Reviewer 2: ```suggestion
An amdSec will also have one or more sourceMD sections. This is the only
```

Reading below, should it `zero or more` where it looks like it isn't necessarily used for a standard transfer? Also, shall we expand on `sourceMD` to be `source metadata` as in the above approach?

Author: Good point, will clarify and add your suggestion.

non-PREMIS section in an amdSec and no XML schema is used for the contents.
SourceMD is created from the contents of the bag-info.txt whenever a zipped or
unzipped bag transfer type is used. It is also created for a disk image transfer
if the user entered meteadata about the disk image in the Transfer tab.

.. _filesec:

<fileSec>
^^^^^^^^^

There is one fileSec listing all files. The fileSec is organized into two or
more of the following fileGrps:

* original
* preservation
* service
* submissionDocumentation
* metadata
* license
* text/ocr
* deleted

These are known as USE attributes and must accompany each fileGrp. The fileGrp

> original is required for all METS files. If the AIP includes normalized files

Reviewer 2: This is an interesting one. We normalize on two occasions so should we say specifically `normalized for preservation` so we aren't opening a question about `normalized for access`?

Author: good point. Best to be clear!

then the fileGrp preservation is used. The service fileGrp may be used if the

> AIP contains the subfolder (e.g. as the output of digitization workflows).

Reviewer 2: ```suggestion
AIP contains the sub-folder (e.g. as the output of digitization workflows).
```

> Each fileGrp must also have a GROUPID and an ID. The GROUPID is the same UUID as

Reviewer 2: I wonder if we can expand on their uses here? So ID is used to link our structmap and our premis object identifier (I believe). GROUP ID I actually don't know... I think this information is useful for tech uses too, we always want to know how we're connecting the relationships across sections in the METS file to bring information about files together so it is truly machine usable and it helps us explain that too to non-tech people where METS is otherwise quite a foreboding looking document! :slightly_smiling_face:

Author: I've tried to clean it up after doing a bit of digging as you know what, it was immediately clear to me either! Will have to have you review the fileSec section to see if it makes sense now.

the ID. Related files will have the same GROUPID.  Below is an example of the
original fileGrp section in the AIP METS file:

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 8830-8850

.. _structmap:

<structMap>
^^^^^^^^^^^

There will be at least one structMap in the METS. It is labeled "Archivematica

> Default" and lists the directories and files in the objects directory.

Reviewer 2: How about:

```
There will be at least one physical structMap in the METS. It is labeled 
"Archivematica Default" and lists the directories and files in the objects 
directory as they are laid out on disk.
```

To place it in contrast with `logical` below.

Author: I do think that's much clearer, thank you!

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 8903-8919

If the *document empty directories* microservice runs, then a second structMap
is created. It is labeled "Normative Directory Structure." At AIP re-ingest the

> new logical structMap will be parsed to re-create the empty directories.

Reviewer 2: This is nice, I'm just wondering about teasing apart logical and physical, so I was thinking of something along the lines of: 
```
is created. It is labeled "Normative Directory Structure." At AIP re-ingest this
new 'logical' structMap will be parsed to re-create the empty directories.

A logical structmap describes a relationship between objects in an Archivematica
AIP that might not be true to their 'physical' layout on disk, as is here where 
empty directories have previously been removed but are still recorded so as to
be recreated in the future.
```
What do you think?

Author: Yes, I think this makes the difference even clearer for me!

> If a user includes a custom structMap in the transfer, then this will also be

Reviewer 2: Shall we link out to the custom structMap section? [ref](https://www.archivematica.org/en/docs/archivematica-1.10/user-manual/transfer/import-metadata/#structmap-xml).

Author: excellent idea, will add that in! I even referred to it when writing.

uploaded with a transfer. Below is an example of a PREMIS Rights statement in
rightsMD section of the AIP METS file:

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 1794-1829

An amdSec will also have zero or more source metadata, or sourceMD, sections.
This is the only non-PREMIS section in an amdSec and no XML schema is used for
the contents. SourceMD is created from the contents of the bag-info.txt whenever
a zipped or unzipped bag transfer type is used. It is also created for a disk
image transfer if the user entered meteadata about the disk image in the
Transfer tab.

.. _filesec:

<fileSec>
^^^^^^^^^

There is one fileSec listing all files. The fileSec is organized into two or
more of the following fileGrps:

* original
* preservation
* service
* submissionDocumentation
* metadata
* license
* text/ocr
* deleted

These are known as USE attributes and must accompany each fileGrp. The fileGrp
original is required for all METS files. If the AIP includes normalized for
preservation files then the fileGrp preservation is used. The service fileGrp
may be used if the AIP contains the sub-folder (e.g. as the output of

> digitization workflows).

Reviewer: A bit of styling here on the fileGrp names might be good for clarity, e.g.:

> The fileGrp *original* is required for all METS files.  If the AIP includes normalized for
preservation files then the fileGrp *preservation* is used. The *service* fileGrp
may be used if the AIP contains the sub-folder (e.g. as the output of
digitization workflows).

(Doesn't need to be italics, but that gives you an idea of what I mean.)

Each fileGrp will have a GROUPID used to relate different versions of files and

> an ID used later as the FILEID to indentify the file in the Structural Map (see

Reviewer: ```suggestion
an ID used later as the FILEID to identify the file in the Structural Map (see
```

> below).

Reviewer: Instead of `(see below)`, just provide a link to the structmap section:

```
identify the file in the :ref:`Structural Map <structMap>`.
```

The GROUPID includes the UUID (from the objectIdentifierValue) of the original
file. The ID is the UUID (from the objectIdentifierValue) of the file being
described. This means original files will have the same GROUPID and ID. However,
related files (such as those normalized for preservation) will have the same
GROUPID as the original file but a different ID.

Below is an example of the original fileGrp section in the AIP METS file:

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 6383-6403

.. _structmap:

<structMap>
^^^^^^^^^^^

There will be at least one physical Structural Map, or structMap, in the METS.
It is labeled "Archivematica Default" and lists the directories and files in the
objects directory as they are laid out on disk.

.. literalinclude:: scripts/METS.xml
   :language: xml
   :lines: 6456-6472

> If the *document empty directories* microservice runs, then a second structMap

Reviewer: You could clarify here that this is a processing config choice, e.g. "If you have chosen to run the job *document empty directories*..."
